src/databricks/sql/client.py

```diff
@@ -297,6 +298,58 @@ def _check_not_closed(self):
         if not self.open:
             raise Error("Attempting operation on closed cursor")
 
+    def _handle_staging_operation(self):
+        """Make HTTP request using instructions provided by server"""
+
+        row = self.active_result_set.fetchone()
+
+        # TODO: Handle headers. What format will gateway send? json? plaintext?
+        operation, presigned_url, local_file, headers = (
+            row.operation,
+            row.presignedUrl,
+            row.localFile,
+            None,
+        )
+
+        operation_map = {
+            "PUT": requests.put,
+            "GET": requests.get,
+        }
+
+        if operation not in operation_map:
+            raise Error(
+                "Operation {} is not supported. Supported operations are {}".format(
+                    operation, ",".join(operation_map.keys())
+                )
+            )
+
+        req_func = operation_map[operation]
+
+        if local_file:
+            raw_data = open(local_file, "rb")
+        else:
+            raw_data = None
+
+        rq_func_args = dict(url=presigned_url, data=raw_data)
```

Comment:
may need to add headers here

Reply:
Yep, just need to know what format the headers will arrive in so I can reformat them in the way `requests` expects.